src/SFA.DAS.AssessorService.Web/Controllers/Apply/StandardController.cs

```diff
@@ -251,39 +251,26 @@ public async Task<IActionResult> OptInPost(Guid id, string standardReference, st
-
+            if (org == null)
+                return RedirectToAction("Applications", "Application");
 
-            bool optInFollowingWithdrawal = false;
+            IEnumerable<StandardVersion> standards = await _standardVersionApiClient.GetStandardVersionsByIFateReferenceNumber(standardReference);
+            StandardVersion stdVersion = standards.FirstOrDefault(x => x.Version.Equals(version, StringComparison.InvariantCultureIgnoreCase));
+            if (stdVersion == null)
+                return RedirectToAction("Applications", "Application");
 
-            //Get all previous applications for standard and find application that contains version in question
-            var previousApplications = await _applicationApiClient.GetPreviousApplicationsForStandard(application.OrganisationId, standardReference);
-            if (previousApplications != null)
-            {
-                //Most recent application must not be a withdrawal
-                var mostRecentApplication = previousApplications.FirstOrDefault();
-                if (mostRecentApplication.ApplicationType != ApplicationTypes.Withdrawal && mostRecentApplication.ApplyData.Apply.Versions.Contains(version))
-                {
-                    //Get all previously withdrawn applications for standard and check to see if version in question has been withdrawn
-                    var previousWithdrawnApplicationsForStandard = await _applicationApiClient.GetAllWithdrawnApplicationsForStandard(application.OrganisationId, stdVersion.LarsCode);
-                    if (previousWithdrawnApplicationsForStandard != null)
-                    {
-                        if (previousWithdrawnApplicationsForStandard.Where(x => x.ApplyData.Apply.Versions.Contains(version) && x.StandardApplicationType == StandardApplicationTypes.VersionWithdrawal).Any())
-                        {
-                            optInFollowingWithdrawal = true;
-                        }
-                    }
-                }
-            }
+            IEnumerable<AppliedStandardVersion> appliedVersions = await _orgApiClient.GetAppliedStandardVersionsForEPAO(org?.OrganisationId, standardReference);
+            AppliedStandardVersion appliedVersion = appliedVersions.FirstOrDefault((x => x.Version.Equals(version, StringComparison.InvariantCultureIgnoreCase)));
+            if (appliedVersion == null || appliedVersion.ApprovedStatus == "Approved" || appliedVersion.ApprovedStatus == "Apply in progress" || appliedVersion.ApprovedStatus == "Feedback Added")
+                return RedirectToAction("Applications", "Application");
 
-            await _orgApiClient.OrganisationStandardVersionOptIn(id, contact.Id, org.OrganisationId, standardReference, version, stdVersion.StandardUId, optInFollowingWithdrawal, $"Opted in by EPAO by {contact.Username}");              
+            DateTime? effectiveTo = appliedVersion.StdVersionEffectiveTo;
+            bool optInFollowingWithdrawal = effectiveTo.HasValue;
```

Comment:
Can these lines be combined?

bool optInFollowingWithdrawal = appliedVersion.StdVersionEffectiveTo.HasValue 

Don't know if it's worth adding appliedversion.StdVersionEffectiveTo < DateTime.UtcNow.

There are cases where the effective to is set in the future, I'd hope that scenario wouldn't impact here as you shouldn't be able to opt in to a version that you are already approved for, which probably feeds into the GetAppliedVersions logic so ignore this bit. Just typing out loud.